### PR TITLE
Simple Options workshop documentation

### LINKED ISSUE #100: Writen tutorial for the simple options example
Based on the videos, a written guide to how to start with cosmwasm-template and build your first contract from scratch.

Intro (defining task): https://vimeo.com/457486858

Setup (most is already in docs, you can add the cosmwasm-template part): https://vimeo.com/457712351

Writing contract code: https://vimeo.com/457702442

Writing tests (and json schema): https://vimeo.com/457705991

### PR DESCRIPTION
**Pull request by orkunkl**
First Merge https://github.com/CosmWasm/cosmwasm-examples/pull/92 
Resolves https://github.com/CosmWasm/docs/issues/100

### CODE REVIEW

**Comment by orkunkl** [comment]
First iteration. I will go over it once more.

File: learn/simple-option/develop.md [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+    Burn {},
+}
+```
+
+::: tip
+Canonical and Human Addresses
+Canonical Addresses represent binary format of crypto addresses.
+Human Addresses on the other hand are great for the UI. They are always a subset of ascii text, and often contain security checks - such as chain-prefix in Bech32, e.g. cosmos1h57760w793q6vh06jsppnqdkc4ejcuyrrjxnke
+
+```canonicalize(humanize(canonical_addr)) == canonical_addr```
```
Comment (ethanfrey) [inline learn/simple-option/develop.md]: Just single ` at beginning and end I think.

File: learn/simple-option/setup.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Assuming you have a followed section above, then the following should get you a new repo to start a contract:
+
+First, install **cargo-generate**. Unless you did that before, run this line now:
+
+```cargo install cargo-generate --features vendored-openssl```
+
+Now, use it to create your new contract. Go to the folder in which you want to place it and run:
+
+```cargo generate --git https://github.com/CosmWasm/cosmwasm-template.git --name simple-option```
```
Comment (ethanfrey) [inline learn/simple-option/setup.md]: I think the triple ticks need to be on a line by themself. And just us a single tick for one line fragments

File: learn/simple-option/develop.md [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+
+```sh
+cargo fmt
+```
+
+Normally Rust compiler does it's job great, leads you to the solution of errors shows warnings etc.
+But we can give the code another round of linting.
+
+```sh
+cargo clippy -- -D warnings
```
Comment (ethanfrey) [inline learn/simple-option/develop.md:399]: Nice one.

But please show how to install clippy

File: learn/simple-option/develop.md [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+RUSTFLAGS='-C link-arg=-s' cargo wasm
+```
+
+Reproducible and optimized compilation:
+
+```sh
+docker run --rm -v "$(pwd)":/code \
+  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+  cosmwasm/rust-optimizer:0.9.0
```
Comment (ethanfrey) [inline learn/simple-option/develop.md]: Let's use 0.10.3, no?

File: learn/simple-option/testing.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+For each test, test specific variables such as block time, state must be mocked. Write a function for easy setup.
+
+```rust
+#[test]
+fn proper_initialization() {
+    /*
+     * mock dependencies
+     * first input is canonical address length
+     * and the second is the initial fund in the contract
+     */
```
Comment (ethanfrey) [inline learn/simple-option/testing.md:39]: Missing array closure. Or my renderer is broken

**Review by ethanfrey — APPROVED** [review]
Looks good. Some minor comments

**Comment by orkunkl** [comment]
There are some code references to example/simple-option master branch which is under development.
- I will merge https://github.com/CosmWasm/cosmwasm-examples/pull/95 and cut a release
- Change simple-option master branch references to the tag
- And then merge this branch

**Pull request merged**